lib/templates/keyfns.ts

```diff
@@ -74,14 +74,16 @@ export function inputParamKeyFn(slot : InputParamSlot) {
 
 export function paramKeyFn(slot : ParamSlot) {
     const id = slot.schema.getArgument('id');
+    const symmetric = slot.schema.getArgument(slot.name)?.getImplementationAnnotation('symmetric');
```

Comment:
If you can make this typesafe (ish) if you do `getImplementationAnnotation<boolean>('symmetric')`